pywr/parameters/_thresholds.pyx

```diff
@@ -187,6 +187,98 @@ cdef class NodeThresholdParameter(AbstractThresholdParameter):
 NodeThresholdParameter.register()
 
 
+cdef class MultipleThresholdIndexParameter(IndexParameter):
+    """ Multiple threshold holder returning an index based on the previous flow in a node
+    against the given thresholds.
+
+    Parameters
+    ----------
+    node : `pywr.core.AbstractNode`
+    thresholds : iterable of `Parameter` instances or floats
+    """
+    def __init__(self, model, node, thresholds, use_max_flow=False, **kwargs):
+        super(MultipleThresholdIndexParameter, self).__init__(model, **kwargs)
+        self.node = node
+        self.thresholds = thresholds
+
+    cpdef int index(self, Timestep timestep, ScenarioIndex scenario_index) except? -1:
+        """Returns the index of the first threshold the node flow is above
+
+        The index is zero-based. For example, if only one threshold is
+        supplied then the index is either 0 (above) or 1 (below). For two
+        thresholds the index is either 0 (above both), 1 (in between), or 2 (below
+        both), and so on.
+        """
+        cdef double flow
+        cdef int index, j
+
+        flow = self.node._prev_flow[scenario_index.global_id]
+        index = len(self.thresholds)
+        for j, threshold in enumerate(self.thresholds):
+            if isinstance(threshold, (float, int)):
+                target_threshold = threshold
+            else:
+                target_threshold = threshold.get_value(scenario_index)
+            if flow >= target_threshold:
+                index = j
+                break
+        return index
+
+    @classmethod
+    def load(cls, model, data):
+        node = model.nodes[data.pop("node")]
+        thresholds = [load_parameter(model, d) for d in data.pop("thresholds")]
+        return cls(model, node, thresholds, **data)
+MultipleThresholdIndexParameter.register()
+
+
+cdef class MultipleThresholdParameterIndexParameter(IndexParameter):
+    """ Multiple threshold holder returning an index based on the value in the parameter
+    against the given thresholds.
+
+    Parameters
+    ----------
+    parameter : Parameter
+    thresholds : iterable of `Parameter` instances or floats
+    """
+    def __init__(self, model, parameter, thresholds, use_max_flow=False, **kwargs):
+        super(MultipleThresholdParameterIndexParameter, self).__init__(model, **kwargs)
+        self.parameter = parameter
+        self.thresholds = thresholds
+
+    cpdef int index(self, Timestep timestep, ScenarioIndex scenario_index) except? -1:
+        """Returns the index of the first threshold the node flow is above
+
+        The index is zero-based. For example, if only one threshold is
+        supplied then the index is either 0 (above) or 1 (below). For two
+        thresholds the index is either 0 (above both), 1 (in between), or 2 (below
+        both), and so on.
+        """
+        cdef double flow
+        cdef int index, j
+
+        flow = self.parameter.value(timestep, scenario_index)
```

Comment:
This is a generic value now, not a "flow".

It should also be `.get_value(scenario_index)`

Reply:
when i put this to `get_value(scenario_index)` it returns the value from the previous day i think whereas this gives the current day which is what i was after

Reply:
This is most likely due to `self.parameter` not being registered as a child of self. See comment I just added. It means that `self.parameter` is not evaluated before `self`, and hence giving you the old value.